lab4/list_operations.py

```diff
@@ -0,0 +1,55 @@
+def move_elements(i, array):
+    array.append(array[len(array) - 1])
+    for i in range(len(array) - 2, i, -1):
+        array[i] = array[i - 1]
+    array.remove(array[len(array) - 1])
+
+
+def delete_elements(i, j, array):
+    for el in range(i, j + 1):
+        array.remove(array[el])
+
+
+def _replace(replaced_real, replaced_imaginary, new_real, new_imaginary, array):
+    for i in array:
+        if i[0] == replaced_real and i[1] == replaced_imaginary:
+            i[0] = new_real
+            i[1] = new_imaginary
+            i[2] = int(math.sqrt(new_real * new_real + new_imaginary * new_imaginary))
+
+
+def _sum(start, end, array):
+    sum_real = 0
+    sum_imaginary = 0
+    if end is not len(array):
+        end += 1
+    for i in range(start, end):
+        sum_real += array[i][0]
+        sum_imaginary += array[i][1]
+
+    print(sum_real, end=" ")
+    print(sum_imaginary)
+
+
+def product(start, end, array):
+    prod_real = prod_imaginary = 0
+    real = imaginary = 0
+
+    for i in range(start, end + 1):
+        if i == start:
+            real = array[i][0]
+            imaginary = array[i][1]
+        else:
+            prod_real += array[i][0] * real - imaginary * array[i][1]
+            prod_imaginary += real * array[i][1] + imaginary * array[i][0]
+            real = prod_real
+            imaginary = prod_imaginary
+
+    print(prod_real, end=" ")
+    print(prod_imaginary, end=" ")
+
+
+def sort_descending_imaginary(array):
```

Comment:
lets not mix functionalities, lets say you want to reuse this function, well... you can't. and that's because you added that extra print()

functions should do only one task and that task only
your function should only manipulate the array, and not print it